runtime/jcl/common/thread.cpp

```diff
@@ -280,15 +280,20 @@ Java_java_lang_Thread_interruptImpl(JNIEnv *env, jobject rcv)
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 	j9object_t receiverObject = J9_JNI_UNWRAP_REFERENCE(rcv);
 	J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, receiverObject);
-	if (J9VMJAVALANGTHREAD_STARTED(currentThread, receiverObject)) {
-		if (NULL != targetThread) {
+	if (NULL != targetThread) {
+		if (J9VMJAVALANGTHREAD_STARTED(currentThread, receiverObject)) {
 			void (*sidecarInterruptFunction)(J9VMThread*) = vm->sidecarInterruptFunction;
 			if (NULL != sidecarInterruptFunction) {
 				sidecarInterruptFunction(targetThread);
 			}
 			omrthread_interrupt(targetThread->osThread);
+		} else if (J2SE_VERSION(vm) >= J2SE_V14) {
+			J9VMJAVALANGTHREAD_SET_DEADINTERRUPT(currentThread, receiverObject, JNI_TRUE);
 		}
+	} else if (J2SE_VERSION(vm) >= J2SE_V14) {
```

Comment:
`#if JAVA_SPEC_VERSION >= 14` can be used instead of a runtime check.

Reply:
Replaced with `#if JAVA_SPEC_VERSION >= 14` as suggested above.